### ISSUE TITLE
Implement `__repr__` for `cupyx.profiler._time._PerfCaseResult`

### ISSUE THREAD

**Issue by kmaehashi** [cupy/cupy#6319]
### Description

Let's implement `__repr__` for interactive benchmarking.

```
>>> cupyx.profiler.benchmark(lambda: numpy.asarray_chkfinite(z), n_repeat=10, n_warmup=0)
<cupyx.profiler._time._PerfCaseResult object at 0x7f3025808c40>
```

**Comment by Subhraneel77** [comment]
Can I Give It A Try?

**Comment by kmaehashi** [comment]
@Subhraneel77 Sure!

**Comment by HallowedDust5** [comment]
I want to contribute - is there any format that the returned string should have?

**Comment by kmaehashi** [comment]
Fixed in #6617
